### PR TITLE
Non utf-8 data restriction

### PR DESCRIPTION
**Pull request by simaranjit**
We are suffering from a weird problem where one of our doc in bulk update has some non-utf8 characters and when that data goes to ES for indexing it throws NullPointerException at ES level, because `JSON::stringify($source, JSON_UNESCAPED_UNICODE);` returns false which ideally skips `doc` part and makes packet like this

```
{ "update" : {"_id" : "1", "_type" : "type1", "_index" : "index1"} }
{ "doc" : {"field2" : "value2"} }
{ "update" : {"_id" : "2", "_type" : "type2", "_index" : "index2"} }

{ "update" : {"_id" : "1", "_type" : "type1", "_index" : "index1"} }
{ "doc" : {"field2" : "value2"} }
```

Note: This it not the exact packet, I can't share it since it's around 70MB

Trying to push that packet directly to ES using postman returns this.

```
{
  "error": {
    "root_cause": [
      {
        "type": "null_pointer_exception",
        "reason": null
      }
    ],
    "type": "null_pointer_exception",
    "reason": null
  },
  "status": 500
}
```

It should be handled at coding level, so that it doesn't accept invalid data.

### CODE REVIEW

File: lib/Elastica/Bulk/Action.php [hunk 1]
```diff
@@ -208,7 +209,14 @@ public function toString()
                 $docAsUpsert = (isset($source['doc_as_upsert'])) ? ', "doc_as_upsert": '.$source['doc_as_upsert'] : '';
                 $string .= '{"doc": '.$source['doc'].$docAsUpsert.'}';
             } else {
-                $string .= JSON::stringify($source, JSON_UNESCAPED_UNICODE);
+                $data = JSON::stringify($source, JSON_UNESCAPED_UNICODE);
```
Comment (ruflin) [inline lib/Elastica/Bulk/Action.php]: Can we add a test for this to confirm it works as expected?

File: lib/Elastica/Bulk/Action.php [hunk 2]
```diff
@@ -208,7 +209,14 @@ public function toString()
...
+				if ($data === false) {
+					throw new InvalidException(sprintf(
+												   'Invalid utf-8 data provided: "%s"',
```
Comment (ruflin) [inline lib/Elastica/Bulk/Action.php]: can we get this on one line even though it becomes a little bit long?

Comment (simaranjit) [inline lib/Elastica/Bulk/Action.php]: Sure, I will update it.

File: lib/Elastica/Bulk/Action.php (hunk 1 above)
Comment (simaranjit) [inline lib/Elastica/Bulk/Action.php]: Yes, I will add test for it.

**Comment by simaranjit** [comment]
@ruflin Thanks, I will do the required changes.

**Comment by simaranjit** [comment]
Hi @ruflin looks like build system is down?

**Comment by ruflin** [comment]
@simaranjit Seems like Travis had some issues. Can you push a new commit and see if runs?

**Comment by simaranjit** [comment]
Hi @ruflin 

Unfortunately, no luck
```The command "sudo apt-get -o Dpkg::Options::="--force-confnew" install -y docker-engine=1.13.1-0~ubuntu-trusty" failed and exited with 100 during .```

**Comment by p365labs** [comment]
@simaranjit  I'm testing PR #1332. This will allow to patch the error on travis build

**Comment by ruflin** [comment]
@simaranjit #1332 is merged. Can you rebase on top?

**Comment by sukhcharanvirdi** [comment]
Thank you @ruflin I am on it.

**Comment by ruflin** [comment]
@simaranjit Sorry that we never managed to complete this.

**Comment by simaranjit** [comment]
@ruflin yeah I'm sorry for that. I will probably create a new PR. It has been so long and I got little out of context. Will re check and open the new PR. Does it sound good?

**Pull request closed without merging**